src/pages/storage/forms/DuplicateVolumeForm.tsx

```diff
@@ -0,0 +1,279 @@
+        .then((operation) => {
+          toastNotify.info(`Duplication of volume ${volume.name} started.`);
+          eventQueue.set(
+            operation.metadata.id,
+            () => notifySuccess(values.name, values.project, values.pool),
+            (msg) =>
+              toastNotify.failure(
+                "Volume duplication failed.",
+                new Error(msg),
+                existingVolumeLink,
+              ),
+          );
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            "Volume duplication failed.",
+            e,
+            existingVolumeLink,
+          );
+        })
+        .finally(() => {
+          close();
+        });
+    },
+  });
+
+  const disableClusterMemberSelector =
+    formik.values.project !== volume.project
+      ? "Can not migrate cluster member when changing project"
+      : undefined;
+
+  const disableProjectSelector =
+    formik.values.member !== volume.location
+      ? "Can not change project when migrating cluster member"
+      : undefined;
+
+  return (
+    <Modal
+      close={close}
+      className="duplicate-instances-modal"
+      title="Duplicate volume"
+      buttonRow={
+        <>
+          <Button
+            appearance="base"
+            className="u-no-margin--bottom"
+            type="button"
+            onClick={close}
+          >
+            Cancel
+          </Button>
+          <ActionButton
+            appearance="positive"
+            className="u-no-margin--bottom"
+            loading={formik.isSubmitting}
+            disabled={!formik.isValid || projectsLoading || volumesLoading}
+            onClick={() => void formik.submitForm()}
+          >
+            Duplicate
+          </ActionButton>
+        </>
+      }
+    >
+      <Form onSubmit={formik.handleSubmit}>
+        <Input
+          {...formik.getFieldProps("name")}
+          type="text"
+          label="New volume name"
+          error={formik.touched.name ? formik.errors.name : null}
+        />
+        <ClusterMemberSelector
+          {...formik.getFieldProps("member")}
+          id="member"
+          label="Target cluster member"
+          disableReason={disableClusterMemberSelector}
+        />
+        <StoragePoolSelector
+          project={volume.project}
+          value={formik.values.pool}
+          setValue={(value) => void formik.setFieldValue("pool", value)}
+          selectProps={{
+            id: "pool",
+            label: "Storage pool",
+          }}
+        />
+        <Select
+          {...formik.getFieldProps("project")}
+          id="project"
+          label="Target project"
+          options={projects.map((project) => {
+            return {
+              label: project.name,
+              value: project.name,
+            };
+          })}
+          disabled={!!disableProjectSelector}
+          help={disableProjectSelector}
+        />
```

Comment:
Shall we remove the target project selector to keep this form a bit simpler? Especially, the exclusive disable logic is hard to gasp for the user.

Reply:
sure I'm fine with that, let's then add a comment here with reference to the upstream bug issue so we can have a reminder to resolve this in the future.

Reply:
actually, on second thought, I think duplication to a different project could be a common use case. Also the project selector is available for instance duplication as well, I think we should keep the interaction somewhat consistent?  Maybe we only remove the project selector for the clustered case?

Reply:
Let's keep it in and simplify once the bug is fixed upstream